pkg/netutils/ips.go

```diff
@@ -4,51 +4,91 @@ import (
 	"fmt"
 	"net"
 	"strings"
-
-	"github.com/sirupsen/logrus"
 )
 
-// GetDefaultIPNet returns the default interface for the node, and the subnet mask for that node, using the same logic
-// as k0s in https://github.com/k0sproject/k0s/blob/v1.30.4%2Bk0s.0/internal/pkg/iface/iface.go#L61
-func GetDefaultIPNet() (*net.IPNet, error) {
-	ifs, err := net.Interfaces()
+// FirstValidAddress returns the first found non-local IPv4 address
+// that's not part of pod network for the given interface.
```

Comment:
This function doesn't care about pod networking at all, that's an interface name filter and this takes an interface name

Reply:
it filters interfaces based on whether or not they have valid ip nets which takes pod networking into account though